src/hotspot/share/utilities/exceptions.cpp

```diff
@@ -144,14 +150,19 @@ void Exceptions::_throw(JavaThread* thread, const char* file, int line, Handle h
   ResourceMark rm(thread);
   assert(h_exception() != nullptr, "exception should not be null");
 
-  // tracing (do this up front - so it works during boot strapping)
-  // Note, the print_value_string() argument is not called unless logging is enabled!
-  log_info(exceptions)("Exception <%.*s%s%.*s> (" PTR_FORMAT ") \n"
-                       "thrown [%s, line %d]\nfor thread " PTR_FORMAT,
-                       MAX_LEN, h_exception->print_value_string(),
-                       message ? ": " : "",
-                       MAX_LEN, message ? message : "",
-                       p2i(h_exception()), file, line, p2i(thread));
+  if (log_is_enabled(Info, exceptions)) {
+    // tracing (do this up front - so it works during boot strapping)
+    // Note, the print_value_string() argument is not called unless logging is enabled!
+    log_info(exceptions)("Exception <%.*s%s%.*s> (" PTR_FORMAT ") \n"
```

Comment:
The way `log_info` et al. work already implicitly checks if the logging is enabled and otherwise it doesn't evaluate anything else, so there is no need to make this change here. In the previous case there was some extra code that is now no longer executed unless needed so that was okay. The key point is that in general you do not need to guard simple `log_xxx(foo)(...)` statements with `log_is_enabled`.